### DISCUSSION TITLE
Mncontour

### DISCUSSION THREAD

**Discussion by sumankumbhakar** [scikit-hep/iminuit#788] [Q&A]
I am trying to get the contour plots for different confidence levels, But it is giving me following error

E MnContours unable to find first two points
E MnContours unable to find first two points

I dont get why so?

**Comment by HDembinski** [comment]
Without more information, it is impossible to say. Check that your fit converged and that computing the Hesse matrix succeeded. Most likely you are trying to fit a function that has no minimum, which contains perfectly correlated parameters or which is degenerate (you have parameters not constrained by the data, so that the derivative along that parameter is flat).

**Comment by sumankumbhakar** [comment]
Thanks for your quick reply. Just to give you more information, I am getting the following output after running the program

┌─────────────────────────────────────────────────────────────────────────┐
│                                Migrad                                   │
├──────────────────────────────────┬──────────────────────────────────────┤
│ FCN = 10.95                      │              Nfcn = 253              │
│ EDM = 2.56e-14 (Goal: 0.0002)    │                                      │
├──────────────────────────────────┼──────────────────────────────────────┤
│          Valid Minimum           │        No Parameters at limit        │
├──────────────────────────────────┼──────────────────────────────────────┤
│ Below EDM threshold (goal x 10)  │           Below call limit           │
├───────────────┬──────────────────┼───────────┬─────────────┬────────────┤
│  Covariance   │     Hesse ok     │ Accurate  │  Pos. def.  │ Not forced │
└───────────────┴──────────────────┴───────────┴─────────────┴────────────┘
┌───┬──────┬───────────┬───────────┬────────────┬────────────┬─────────┬─────────┬───────┐
│   │ Name │   Value   │ Hesse Err │ Minos Err- │ Minos Err+ │ Limit-  │ Limit+  │ Fixed │
├───┼──────┼───────────┼───────────┼────────────┼────────────┼─────────┼─────────┼───────┤
│ 0 │ x    │ -0.054e-6 │ 0.041e-6  │ -0.063e-6  │  0.034e-6  │         │         │       │
│ 1 │ y    │ -0.51e-3  │  0.24e-3  │  -0.24e-3  │  0.24e-3   │         │         │       │
└───┴──────┴───────────┴───────────┴────────────┴────────────┴─────────┴─────────┴───────┘
┌──────────┬───────────────────────┬───────────────────────┐
│          │           x           │           y           │
├──────────┼───────────┬───────────┼───────────┬───────────┤
│  Error   │ -0.063e-6 │ 0.034e-6  │ -0.24e-3  │  0.24e-3  │
│  Valid   │   True    │   True    │   True    │   True    │
│ At Limit │   False   │   False   │   False   │   False   │
│ Max FCN  │   False   │   False   │   False   │   False   │
│ New Min  │   False   │   False   │   False   │   False   │
└──────────┴───────────┴───────────┴───────────┴───────────┘
Covariance matrix = 
┌───┬─────────────────────┐
│   │         x         y │
├───┼─────────────────────┤
│ x │  1.66e-15 -5.87e-12 │
│ y │ -5.87e-12  5.56e-08 │
└───┴─────────────────────┘

correlation matrix=
┌───┬─────────────┐
│   │     x     y │
├───┼─────────────┤
│ x │     1 -0.61 │
│ y │ -0.61     1 │
└───┴─────────────┘

E MnContours unable to find first two points
E MnContours unable to find first two points
Traceback (most recent call last):
  File "/Users/suman/Documents/Ymutau/Y-tau-e.py", line 55, in <module>
    pls=m.draw_mncontour("x","y",cl=(0.68,0.9,0.99))
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/iminuit/minuit.py", line 1894, in draw_mncontour
    cs = ContourSet(plt.gca(), c_val, c_pts, codes)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/matplotlib/contour.py", line 812, in __init__
    kwargs = self._process_args(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/matplotlib/contour.py", line 1038, in _process_args
    points = np.concatenate(flatseglist, axis=0)
  File "<__array_function__ internals>", line 180, in concatenate
ValueError: all the input arrays must have same number of dimensions, but the array at index 0 has 2 dimension(s) and the array at index 1 has 1 dimension(s)
